src/celeritas/UnitTypes.hh

```diff
@@ -0,0 +1,291 @@
+    }
+    static char const* label() { return "amu"; }
+};
+
+struct Barn
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value() { return units::barn; }
+    static char const* label() { return "b"; }
+};
+
+struct Millibarn
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return real_type(1e-3) * units::barn;
+    }
+    static char const* label() { return "mb"; }
+};
+
+//!@}
+//---------------------------------------------------------------------------//
+//!@{
+//! \name Gaussian units
+
+struct Centimeter
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return units::centimeter;
+    }
+    static char const* label() { return "cm"; }
+};
+
+struct Gram
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value() { return units::gram; }
+    static char const* label() { return "g"; }
+};
+
+struct Gauss
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value() { return units::gauss; }
+    static char const* label() { return "G"; }
+};
+
+//! Inverse cubic centimeter for number densities
+struct InvCentimeterCubed
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return 1 / (units::centimeter * units::centimeter * units::centimeter);
+    }
+    static char const* label() { return "1/cm^3"; }
+};
+
+//! Molar density
+struct MolPerCentimeterCubed : UnitProduct<Mol, InvCentimeterCubed>
+{
+    static char const* label() { return "mol/cm^3"; }
+};
+
+//!@}
+//---------------------------------------------------------------------------//
+//!@{
+//! \name SI units
+
+struct Meter
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value() { return units::meter; }
+    static char const* label() { return "m"; }
+};
+
+struct Kilogram
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return units::kilogram;
+    }
+    static char const* label() { return "kg"; }
+};
+
+struct Second
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value() { return units::second; }
+    static char const* label() { return "s"; }
+};
+
+struct Tesla
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value() { return units::tesla; }
+    static char const* label() { return "T"; }
+};
+
+//!@}
+//---------------------------------------------------------------------------//
+//!@{
+//! \name CLHEP units
+
+struct Millimeter
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return units::millimeter;
+    }
+    static char const* label() { return "mm"; }
+};
+
+struct Nanosecond
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        return units::nanosecond;
+    }
+    static char const* label() { return "ns"; }
+};
+
+//! Unit mass in CLHEP system
+struct ClhepUnitMass
+{
+    static CELER_CONSTEXPR_FUNCTION real_type value()
+    {
+        // Note: order matters in CHLEP system to result in exactly 1 with
+        // floating point arithmetic
+        return constants::e_electron * (kilogram / coulomb) * 1e-6;
```

Comment:
if get 1.0 is important wouldn't the following be more resilient to change in the CPU ieee implementations?
```suggestion
        if constexpr (CELERITAS_UNITS == CELERITAS_UNITS_CLHEP)
           return 1.0 ...;
        else 
           return constants::e_electron * (kilogram / coulomb) * 1e-6;
```

Reply:
Yep, that's probably smarter.